modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -42,7 +42,8 @@
     protected String groupId = "io.swagger";
     protected String artifactId = "swagger-java-client";
     protected String artifactVersion = "1.0.0";
-    protected String sourceFolder = "src/main/java";
+    protected String projectFolder = "src/main";
+    protected String sourceFolder = projectFolder + "/java";
```

Comment:
OK. I'll do it in this PR.
Shouldn't java 7  [Path](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html) be used instead of having  File.separator everywhere ?